audio/audio_hw.c

```diff
@@ -1254,10 +1331,31 @@ static int in_set_parameters(struct audio_stream *stream, const char *kvpairs)
     return 0;
 }
 
-static char * in_get_parameters(const struct audio_stream *stream,
+static char* in_get_parameters(const struct audio_stream *stream,
                                 const char *keys)
 {
-    return strdup("");
+    ALOGV("+in_get_parameters(%p) '%s'", stream, keys);
+
+    struct stream_in_common *in = (struct stream_in_common *)stream;
+    struct str_parms *query = str_parms_create_str(keys);
+    struct str_parms *reply = str_parms_create();
+    char value[BUFFER_SIZE];
+    char *str;
+
+    if (str_parms_get_str(query, AUDIO_PARAMETER_STREAM_SUP_FORMATS,
+        value, sizeof(value)) >= 0) {
+        str_parms_add_audio_format(reply, 
+            AUDIO_PARAMETER_STREAM_SUP_FORMATS, in->format);
+        str = strdup(str_parms_to_str(reply));
```

Comment:
strdup() not necessary. See
https://android.googlesource.com/platform/hardware/libhardware/+/refs/heads/master/modules/usbaudio/audio_hal.c#273

Reply:
Done.